eval.go

```diff
@@ -624,218 +324,402 @@ func newUserTypeInfo(userType reflect.Type) (*userTypeInfo, error) {
+		case []byte:
+			return string(val), false, nil
+		case []string:
+			b, _ := toJson(val)
+			return string(b), false, nil
+		case float32:
+			return strconv.FormatFloat(float64(val), 'g', -1, 64), true, nil
+		case float64:
+			return strconv.FormatFloat(val, 'g', -1, 64), true, nil
+		case int:
+			return strconv.FormatInt(int64(val), 10), true, nil
+		case uint:
+			return strconv.FormatUint(uint64(val), 10), true, nil
+		case int8:
+			return strconv.FormatInt(int64(val), 10), true, nil
+		case uint8:
+			return strconv.FormatUint(uint64(val), 10), true, nil
+		case int16:
+			return strconv.FormatInt(int64(val), 10), true, nil
+		case uint16:
+			return strconv.FormatUint(uint64(val), 10), true, nil
+		case int32:
+			return strconv.FormatInt(int64(val), 10), true, nil
+		case uint32:
+			return strconv.FormatUint(uint64(val), 10), true, nil
+		case int64:
+			return strconv.FormatInt(val, 10), true, nil
+		case uint64:
+			return strconv.FormatUint(val, 10), true, nil
+		case uintptr:
+			return strconv.FormatUint(uint64(val), 10), true, nil
+		case time.Time:
+			return strconv.FormatInt(val.UnixMilli()/1000, 10), true, nil
+		}
+	}
+	return "", false, &userAttrMissingError{attr: attr}
+}
+
+func (t *userTypeInfo) getBytes(v reflect.Value, attr string) ([]byte, bool, error) {
+	info, ok := t.fields[attr]
+	if ok && info.asBytes != nil {
+		result, converted := info.asBytes(v)
+		if len(result) == 0 && isPredefined(attr) {
+			return nil, false, &userAttrMissingError{attr: attr}
+		}
+		return result, converted, nil
+	} else if t.getAttribute != nil {
+		res := t.getAttribute(v, attr)
+		if res == nil {
+			return nil, false, &userAttrMissingError{attr: attr}
+		}
+		switch val := res.(type) {
+		case string:
+			return []byte(val), false, nil
+		case []byte:
+			return val, false, nil
+		case []string:
+			b, _ := toJson(val)
+			return b, false, nil
+		case float32:
+			return strconv.AppendFloat(nil, float64(val), 'g', -1, 64), true, nil
+		case float64:
+			return strconv.AppendFloat(nil, val, 'g', -1, 64), true, nil
+		case int:
+			return strconv.AppendInt(nil, int64(val), 10), true, nil
+		case uint:
+			return strconv.AppendUint(nil, uint64(val), 10), true, nil
+		case int8:
+			return strconv.AppendInt(nil, int64(val), 10), true, nil
+		case uint8:
+			return strconv.AppendUint(nil, uint64(val), 10), true, nil
+		case int16:
+			return strconv.AppendInt(nil, int64(val), 10), true, nil
+		case uint16:
+			return strconv.AppendUint(nil, uint64(val), 10), true, nil
+		case int32:
+			return strconv.AppendInt(nil, int64(val), 10), true, nil
+		case uint32:
+			return strconv.AppendUint(nil, uint64(val), 10), true, nil
+		case int64:
+			return strconv.AppendInt(nil, val, 10), true, nil
+		case uint64:
+			return strconv.AppendUint(nil, val, 10), true, nil
+		case uintptr:
+			return strconv.AppendUint(nil, uint64(val), 10), true, nil
+		case time.Time:
+			return strconv.AppendInt(nil, val.UnixMilli()/1000, 10), true, nil
```

Comment:
As per spec, datetime value is a

> unix timestamp in seconds (**with an optional decimal part**)

```suggestion
			return strconv.AppendFloat(nil, float64(val.UnixMilli())/1000.0, 'f', -1, 64), true, nil
```